libpod/container_internal.go

```diff
@@ -131,14 +133,92 @@ func (c *Container) AttachSocketPath() string {
 	return filepath.Join(c.runtime.ociRuntime.socketsDir, c.ID(), "attach")
 }
 
+// exitFilePath gets the path to the container's exit file
+func (c *Container) exitFilePath() string {
+	return filepath.Join(c.runtime.ociRuntime.exitsDir, c.ID())
+}
+
+// create a bundle path and associated files for an exec session
+func (c *Container) createExecBundle(sessionID string) (err error) {
+	bundlePath := c.execBundlePath(sessionID)
+	if createErr := os.MkdirAll(bundlePath, execDirPermission); createErr != nil {
+		return createErr
+	}
+	defer func() {
+		if err != nil {
+			if err2 := os.Remove(bundlePath); err != nil {
+				logrus.Warnf("error removing exec bundle after creation caused another error: %v", err2)
+			}
+		}
+	}()
+	f, err2 := os.OpenFile(c.execLogPath(sessionID), os.O_RDONLY|os.O_CREATE, execFilePermission)
+	if err2 != nil {
+		err = errors.Wrapf(err2, "Error creating file for container %s exec session %s", c.ID(), sessionID)
+		return
+	}
+	if err2 := f.Close(); err2 != nil {
+		logrus.Warnf("Error in closing newly created log path file: %v", err2)
+	}
+	if err2 := os.MkdirAll(c.execExitFileDir(sessionID), execDirPermission); err2 != nil {
+		// The directory is allowed to exist
+		if !os.IsExist(err2) {
+			err = errors.Wrapf(err2, "error creating OCI runtime exit file path %s", c.execExitFileDir(sessionID))
+		}
+	}
+	return
+}
+
+// cleanup an exec session after its done
+func (c *Container) cleanupExecBundle(sessionID string) error {
+	return os.RemoveAll(c.execBundlePath(sessionID))
+}
+
+// the path to a containers exec session bundle
+func (c *Container) execBundlePath(sessionID string) string {
+	return filepath.Join(c.bundlePath(), sessionID)
+}
+
 // Get PID file path for a container's exec session
 func (c *Container) execPidPath(sessionID string) string {
-	return filepath.Join(c.state.RunDir, "exec_pid_"+sessionID)
+	return filepath.Join(c.execBundlePath(sessionID), "exec_pid")
 }
 
-// exitFilePath gets the path to the container's exit file
-func (c *Container) exitFilePath() string {
-	return filepath.Join(c.runtime.ociRuntime.exitsDir, c.ID())
+// the log path for an exec session
+func (c *Container) execLogPath(sessionID string) string {
+	return filepath.Join(c.execBundlePath(sessionID), "exec_log")
+}
+
+// the socket conmon creates for an exec session
+func (c *Container) execAttachSocketPath(sessionID string) string {
+	return filepath.Join(c.runtime.ociRuntime.socketsDir, sessionID, "attach")
```

Comment:
I seem to recall us doing some trimming on attach socket paths to make sure they didn't exceed 128 characters for rootless - @giuseppe do you remember this?

Reply:
Yeah I if you notice I replace the container ID with session ID. I would have preferred it be a combination of containerID and sessionID, but it is easier to configure conmon to do it this way. Thus, I am respecting the 128 character limit